Can't build cpthyon on Macos Sierra

My system is Macos Sierra, When I build cpython I get error as blow.

--  While building package cpython-3.4.0 using:
      ~/.stack/setup-exe-cache/x86_64-osx/setup-Simple-Cabal-1.22.5.0-ghc-7.10.3 --builddir=.stack-work/dist/x86_64-osx/Cabal-1.22.5.0 build --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
    Logs have been written to: ./.stack-work/logs/cpython-3.4.0.log

    Configuring cpython-3.4.0...
    Building cpython-3.4.0...
    Preprocessing library cpython-3.4.0...
    c2hs: C header contains errors:

    /usr/include/stdio.h:133: (column 19) [ERROR]  >>> Syntax error !
      The symbol `_close' does not fit here.

--- a/c2hs.cabal
+++ b/c2hs.cabal
@@ -112,7 +112,7 @@ flag base3
 Executable c2hs
     Build-Depends:  base >= 2 && < 5,
                     bytestring,
-                    language-c >= 0.5 && < 0.6,
+                    language-c >= 0.6 && < 0.7,
                     filepath,
                     dlist
 

--- a/src/C2HS/Gen/Bind.hs
+++ b/src/C2HS/Gen/Bind.hs
@@ -2309,7 +2309,7 @@ extractCallingConvention cdecl
     hasStdCallAttr = any isStdCallAttr . funAttrs
 
     funAttrs (CDecl specs declrs _) =
-      let (_,attrs',_,_,_) = partitionDeclSpecs specs
+      let (_,attrs',_,_,_,_) = partitionDeclSpecs specs
        in attrs' ++ funEndAttrs declrs ++ funPtrAttrs declrs
 
     -- attrs after the function name, e.g. void foo() __attribute__((...));

